java/java.lsp.server/vscode/src/panels/SshGuidePanel.ts

```diff
@@ -0,0 +1,132 @@
+        this.panel = vscode.window.createWebviewPanel(
+            SshGuidePanel.viewType,
+            "Oracle Cloud Assets",
+            { viewColumn: vscode.ViewColumn.One, preserveFocus: true },
+            {
+                enableCommandUris: true,
+                enableScripts: true,
+                localResourceRoots: [vscode.Uri.file(path.join(context.extensionPath, SshGuidePanel.webviewsFolder))],
+            }
+        );
+        const iconPath: vscode.Uri = vscode.Uri.file(path.join(context.extensionPath, "images", "Apache_NetBeans_Logo.png"));
+        this.panel.iconPath = {
+            light: iconPath,
+            dark: iconPath,
+        };
+
+        this.setHtml(context);
+
+        this.panel.onDidDispose(
+            () => {
+                this.dispose();
+                removeGuidePropertiesFromStateFor(context, SshGuidePanel.statePropertyPrefix);
+            },
+            null,
+            this.disposables
+        );
+
+        this.panel.onDidChangeViewState(
+            () => {
+                if (this.panel.visible) {
+                    this.setHtml(context);
+                }
+            },
+            null,
+            this.disposables
+        );
+
+        this.panel.webview.onDidReceiveMessage(
+            (message) => {
+                if (message.command === "showSSHGuide") {
+                    const ocid: string = context.globalState.get(`${SshGuidePanel.statePropertyPrefix}-ocid`) || "";
+                    toggleGuideFor(SshGuidePanel.actionType, ocid);
+                }
+            },
+            undefined,
+            this.disposables
+        );
+    }
+
+    private async setHtml(context: vscode.ExtensionContext) {
+        let templateFile = path.resolve(__dirname, "../../webviews/ssh-guide.handlebars");
+        templateFile = await readFile(templateFile, "utf-8");
+        const template = Handlebars.compile(templateFile);
+        const ocid: string | undefined = context.globalState.get<string>(`${SshGuidePanel.statePropertyPrefix}-ocid`);
+
+        this.panel.webview.html = template({
+            cspSource: this.panel.webview.cspSource,
+            publicIp: context.globalState.get<string>(`${SshGuidePanel.statePropertyPrefix}-publicIp`),
+            dummyKeyPathLocation,
+            sshConfigLocation,
+            showSSHGuide: shouldHideGuideFor(SshGuidePanel.actionType, ocid) ? "checked" : "",
+            cssUri: this.panel.webview.asWebviewUri(vscode.Uri.file(path.join(context.extensionPath, SshGuidePanel.webviewsFolder, SshGuidePanel.stylesFolder, "guide.css"))),
+            javascriptUri: this.panel.webview.asWebviewUri(
+                vscode.Uri.file(path.join(context.extensionPath, SshGuidePanel.webviewsFolder, SshGuidePanel.scriptsFolder, "ssh-guide.js"))
+            ),
+        });
+    }
+
+    public dispose() {
+        SshGuidePanel.currentPanel = undefined;
+
+        this.panel.dispose();
+        while (this.disposables.length) {
```

Comment:
nitpick: disposables.forEach; the disposable should not contain an undefined ... but if you want to be sure, you can use `x?.dispose()` and avoid if.